### ISSUE TITLE
Sort messages

### ISSUE THREAD

**Issue by tinybigstudio** [nylas/N1#694]
It would be good to sort messages ...

**Comment by grinich** [comment]
Sort by what?

**Comment by tinybigstudio** [comment]
sort by sender, subject and date, kind of like a file system. Those headings would make it easier to use for some users. Something like this, and where these headlines are links to sort by status (read or unread - maybe use a better word), subject and date. Upon a second click it could reverse the sorting order as in A-Z, Z-A for sender and subject and latest or oldest for date.
![artboard 2x](https://cloud.githubusercontent.com/assets/2094957/11768048/e5a2cc8c-a1b8-11e5-8f22-03ee7bc2d662.png)

**Comment by crypticsymbols** [comment]
Sorting by unread status is a big one for me... Fantastic job so far, by the way!

**Comment by mbilker** [comment]
:+1: for sorting by any field. A header row would be nice to click on and to enable sorting by a certain field. May need to add a button to reset sorting back to reverse chronological order.

**Comment by crypticsymbols** [comment]
Super+Click to sort by multiple columns

**Comment by RgnKjnVA** [comment]
Sort by read status is critical for me. It's the only thing that's 'missing' in N1that I've discovered so far. 

I'm so psyched I stumbled on N1. All the email clients' demise the last couple of years has really opened up an opportunity. Great job!

**Comment by DfKimera** [comment]
I've been hopelessly hopping from mail client to mail client for years in search of something better then stock Gmail, and N1 hits really close! Sorting by read status, however, is truly critical to the way I organise myself, and I won't be able to use it as my everyday mail client without that feature.

**Comment by RgnKjnVA** [comment]
I submitted feedback through the client regarding Sort By or Filter On Read status but haven't heard back. I wonder if their 'inbox zero' ethos is precluding Nylas from offering such a feature. N1 is pretty new, 5 months old, so I'm not freaking out yet but will be watching for this feature with keen interest.

**Comment by crypticsymbols** [comment]
To everyone that's really excited about N1, please do your homework. By default all of your data is archived offsite, on Nylas' server. You can run your own instance but if you are worried about privacy, do not use the default.

**Comment by RgnKjnVA** [comment]
Isn't that always the case? Email users' computers and devices don't have the entirety of their email data stored on them. That would be a prohibitively huge amount of data. Is the implication that Nylas can't/shouldn't be trusted? 

BTW on that topic, I really like the idea of putting encryption keys in the hands of end users so that if a company gets hacked *shrug* so what, they don't have the encryption key for my data. Granted, there are pitfalls to this paradigm shift but could be the next wave with more and more companies demonstrating that they are simply ill-equipped to hold back the tide of hacker activity. That's not a slam on Nylas, just a general point on the state of business in our connected world. Also, would data-hungry content providers (Google) balk at moves like this since it would mean that they themselves lose access to your data? Will be interesting to see what happens going forward. 

http://techcrunch.com/2016/03/01/new-egnyte-product-puts-customers-in-control-of-encryption-keys/

**Comment by beansprout** [comment]
I would love to be able to sort by sender.  Right now the only app I can find that does that is Postbox.  
Is it hard to code that in?  Is that why nobody seems to offer it?  Or is it server heavy?

The sort by sender feature is great for getting rid of emails quickly when you let thousands of inbox emails pile up and you suspect there might be important things there... 

I know you can do a search for individual senders but that takes a lot longer.

**Comment by alimuldal** [comment]
This is a basic feature - I've never used any email client that didn't have some type of "sort" functionality. In particular the ability to sort and/or filter by date/read status are essentials.

**Comment by jackiehluo** [comment]
Another request: https://nylas.zendesk.com/agent/tickets/6138

**Comment by codingjoe** [comment]
I'm a bit confused how this issue isn't resolved yet. I get hundreds of mails and with the pro integration of SalesForce it is vital that I answer all my mail in the order they came in as a queue not a stack.

If there is anything I can do, let me know. I would certainly implement it myself if someone points me the way.

**Comment by La-Technition** [comment]
Yes, this is major. We absolutely need this. Outlook has it. Mail for Mac OS has it. But Nylas doesn't. I am surprised it isn't resolved yet, though. 👍
